src/services/mbs.ts

```diff
@@ -0,0 +1,173 @@
+  tasks: { rpms?: Record<string, MbsTaskResponse> };
+  time_completed?: string;
+  /** Module version. The third component of the NSVC */
+  version: string;
+}
+
+export const mbsTaskSchema = z.object({
+  component: z.string(),
+  id: z.number().optional(),
+  nvr: z.string().optional(),
+  state: z.number(),
+});
+
+export type MbsTask = z.infer<typeof mbsTaskSchema>;
+
+/**
+ * This schema fragment preprocesses the dictionary tasks of a module build
+ * into a form that is more suitable for further processing.
+ *
+ * The structure we receive from MBS looks like
+ *
+ *     ...
+ *     tasks: {
+ *       rpms: {
+ *         component1: {
+ *           nvr: "component1-2.3.4-9.module+el8.4.0+1111+aeb4ae67",
+ *           state: 1,
+ *           task_id: 90909090,
+ *         },
+ *         component2: { ... },
+ *         ...
+ *       }
+ *     }
+ *     ...
+ *
+ * This schema fragment takes the RPM part of the component–task map and turns it
+ * into a list of tasks. It ignores non-RPM tasks that might appear in the response
+ * from MBS. After the schema is applied, the output looks like
+ *
+ *     [
+ *       { component: "component1", nvr: "component1-2.3.4-...", ... },
+ *       { component: "component2", ... },
+ *       ...
+ *     ]
+ */
+const rpmsTasksSchema = z.preprocess((input) => {
+  const tasks = input as MbsModuleBuildResponse['tasks'] | undefined;
+  if (!tasks || !tasks.rpms) return [];
+  return Object.entries(tasks.rpms).map(([component, task]) => ({
+    component,
+    id: task.task_id,
+    nvr: task.nvr,
+    state: task.state,
+  }));
+}, mbsTaskSchema.array());
+
+export const moduleBuildSchema = z.object({
+  /** Module context. The last component of the NSVC */
+  context: z.string(),
+  /** Module build ID */
+  id: z.number(),
+  /** The target Koji tag where the module is built */
+  koji_tag: z.string(),
+  /** Module name. The first component of the NSVC */
+  name: z.string(),
+  /** Build owner name */
+  owner: z.string(),
+  /** Git commit URL */
+  scmurl: z.string(),
+  /** Is this a scratch build? */
+  scratch: z.boolean(),
+  state_name: z.string(),
+  /** Module stream. The second component of the NSVC */
+  stream: z.string(),
+  /** List of Koji tasks comprising the module build */
+  tasks: rpmsTasksSchema,
+  /** Timestamp of when the build was completed. MBS should supply this in ISO 8601 format */
+  time_completed: z.string().optional(),
+  /** Module version. The third component of the NSVC */
+  version: z.string(),
+});
+
+export type ModuleBuild = z.infer<typeof moduleBuildSchema>;
+
+export const queryModuleBuild = async (
+  instanceId: MbsInstance,
+  buildId: number
+): Promise<ModuleBuild> => {
+  const instanceUrl = cfg.mbs[instanceId]?.url;
+  const url = `${instanceUrl}/module-builds/${buildId}`;
+
+  let response: AxiosResponse<any> | undefined;
+  try {
+    response = await axios.get(url, {
+      httpsAgent: new Agent({ rejectUnauthorized: false }),
```

Comment:
Why do we need this? If there is issue in SSL it should be fixed in different way.

Reply:
This looks like it was left over from some testing. Removed.